include/boost/geometry/algorithms/detail/overlay/traversal_switch_detector.hpp

```diff
@@ -64,59 +95,46 @@ struct traversal_switch_detector
         isolation_multiple = 2
     };
 
-    typedef typename boost::range_value<Turns>::type turn_type;
-    typedef typename turn_type::turn_operation_type turn_operation_type;
-    typedef std::set<signed_size_type> set_type;
+    using turn_type = typename boost::range_value<Turns>::type;
+    using set_type= std::set<signed_size_type>;
 
     // Per ring, first turns are collected (in turn_indices), and later
     // a region_id is assigned
     struct merged_ring_properties
     {
-        signed_size_type region_id;
+        signed_size_type region_id{-1};
         set_type turn_indices;
-
-        merged_ring_properties()
-            : region_id(-1)
-        {}
     };
 
     struct connection_properties
     {
-        std::size_t count;
-        // Contains turn-index OR, if clustered, minus-cluster_id
+        std::size_t count{0};
+        // Set with turn-index OR (if clustered) the negative cluster_id
         set_type unique_turn_ids;
-        connection_properties()
-            : count(0)
-        {}
     };
 
-    typedef std::map<signed_size_type, connection_properties> connection_map;
+    // Maps region_id -> properties
+    using connection_map = std::map<signed_size_type, connection_properties>;
 
     // Per region, a set of properties is maintained, including its connections
     // to other regions
     struct region_properties
     {
-        signed_size_type region_id;
-        isolation_type isolated;
+        signed_size_type region_id{-1};
+        isolation_type isolated{isolation_unknown};
```

Comment:
I'm just curious. These are native type and enum, correct? Why have you decided to use braces instead of `=`?
```
signed_size_type region_id = -1;
isolation_type isolated = isolation_unknown;
```

Reply:
Ah, it's my new habit now. Because we do it at the office. But I now remember that you and @mloskot objected to it, and indeed it does not make sense. I'll change it.

Reply:
Updated :heavy_check_mark:

Reply:
Right, I now remember that we talked about this. So this is a habit. I was curious that maybe there is rationale behind it that I'm not aware of. Anyway, thanks!